src/test/scala/chiselTests/ChiselSpec.scala

```diff
@@ -360,3 +360,22 @@ trait Utils {
 
   }
 }
+
+/** Contains helpful function to assert both statements to match, and statements to omit */
+trait MatchesOrOmits {
+  private def matches(lines: List[String], matchh: String): Option[String] = lines.filter(_.contains(matchh)).lastOption
+  private def omits(line:    String, omit:         String): Option[(String, String)] =
+    if (line.contains(omit)) Some((omit, line)) else None
+  private def omits(lines: List[String], omit: String): Seq[(String, String)] = lines.flatMap { omits(_, omit) }
+  def matchesOrOmits(output: String)(matchList: String*)(omitList: String*): Unit = {
+    val lines = output.split("\n").toList
+    val unmatched = matchList.flatMap { m =>
+      if (matches(lines, m).nonEmpty) None else Some(m)
+    }.map(x => s"  > $x was unmatched")
+    val unomitted = omitList.flatMap { o => omits(lines, o) }.map {
+      case (o, l) => s"  > $o was not omitted in ($l)"
+    }
+    val results = unmatched ++ unomitted
+    assert(results.isEmpty, results.mkString("\n"))
+  }
```

Comment:
Mega-nit: isn't this `matchesAndOmits`?

Non-blocking feedback: the multiple parameter lists can be hard to grok, but are necessary to have multiple variadic arguments. Using Arrays and named arguments when using this method may make it more obvious.

Alternatively, just having a FIRRTL trimming/canonicalizing utility and letting users do their own ScalaTest matching is more what I was thinking here.